pkg/api/validation/events.go

```diff
@@ -25,11 +25,20 @@ import (
 // ValidateEvent makes sure that the event makes sense.
 func ValidateEvent(event *api.Event) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	if event.Namespace != event.InvolvedObject.Namespace {
-		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.namespace", event.InvolvedObject.Namespace, "namespace does not match involvedObject"))
-	}
+
 	if !util.IsDNSSubdomain(event.Namespace) {
 		allErrs = append(allErrs, errs.NewFieldInvalid("namespace", event.Namespace, ""))
 	}
+	// TODO: event.InvolvedObject is not versioned. References need to be normalizd to internal when converted from external and vice versa.
+	if event.InvolvedObject.Kind == "Minion" || event.InvolvedObject.Kind == "Node" {
```

Comment:
None - there are now two scopes, "namespaced" and "root".  Nodes are "root" resources which means their namespace is effectively "" always (and the code *should* treat that value in most ingress locations as an error).

----- Original Message -----
> >  	if !util.IsDNSSubdomain(event.Namespace) {
> >  		allErrs = append(allErrs, errs.NewFieldInvalid("namespace",
> >  		event.Namespace, ""))
> >  	}
> > +	// TODO: event.InvolvedObject is not versioned. References need to be
> > normalizd to internal when converted from external and vice versa.
> > +	if event.InvolvedObject.Kind == "Minion" || event.InvolvedObject.Kind ==
> > "Node" {
> 
> What namespace are nodes/minions in right now?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/4396/files#r24871235